feature-libs/smartedit/root/services/smart-edit-init.service.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { SmartEditService } from '@spartacus/smartedit/core';
+import { SmartEditLauncherService } from './smart-edit-launcher.service';
+
+@Injectable({ providedIn: 'root' })
+export class SmartEditInitService {
+  static factory(smartEditInitService: SmartEditInitService): () => void {
+    return (): void => {
+      smartEditInitService.init();
+    };
+  }
+
+  constructor(
+    protected smartEditService: SmartEditService,
+    protected smartEditLauncherService: SmartEditLauncherService
+  ) {}
+
+  init(): void {
+    if (this.smartEditLauncherService.isLaunchedInSmartEdit()) {
+      this.smartEditService.processCmsPage();
+    }
+  }
+}
```

Comment:
```suggestion
export const smartEditServiceInitializerFactory = (smartEditService: SmartEditService, smartEditLauncherService: SmartEditLauncherService) => () => {
 if(smartEditLauncherService.isLaunchedInSmartEdit()) {
   smartEditService.processCmsPage();
 }
}
```

Reply:
I am not fond of this change as it reduces code extensibility and also negatively impacts the testing process.

Reply:
The question is whether such a code is going to be extended. I believe this factory function shouldn't be part of the public API as it may cause many issues if used in the wrong way. The other thing is that it's an uncommon way of creating a factory and I don't know whether I like how the service is used - `SmartEditInitService` has to be passed to `deps` in providers to use the `factory` function that is part of `SmartEditInitService` as well...

Reply:
In addition, the convention that you proposed is new to the project, which may lead to ambiguity when someone will have to add similar logic in the future.